Add slaking documents (#422)
hpcugent/vsc_user_docs: intro-HPC/ch_debug_gent.tex

\chapter{HPC-UGent debug cluster}
\label{ch:debug_ugent}

\section{Purpose}
\label{sec:debug_ugent_pupose}
The purpose of the debug cluster is to run debug jobs. Additionally, it can be also used for courses. 
The debug cluster is overcommited, which means that 6 times more CPU core can be
scheduled than physically exist in the cluster. The cluster performance heavily depends on the
workload, and running jobs can slow down or speed up during they are running.  In the case of a
fully loaded cluster, jobs would run about 6 times slower compared to an empty cluster. 
\\ Due to the restrictions (see section~\ref{subsec:debug_ugent_restrictions}) and sharing the CPU
resorces, jobs on this cluster should normally immediately start. As a consequence,

> typical workload for this cluster are:

Reviewer: ```suggestion
typical workloads for this cluster should be limited to the :
```

\begin{itemize}
  \item  Debugging programs
  \item  Try/debug submit scripts

>   \item  Run small jobs when performance is not an issue (courses, hands-ons, etc.)

Reviewer: ```suggestion
  \item  Run small jobs for which performance is not an issue (courses, hands-ons, etc.)
```

  \item  Run interactive jobs (see chapter~\ref{ch:running-interactive-jobs}) with lots of idle time
         (courses, hads-ons, etc.) 

>   \item  Run Cluster Desktops (see chapter~\ref{ch:web_portal}) whith lots of idle time

Reviewer: ```suggestion
  \item  Run Cluster Desktops (see chapter~\ref{ch:web_portal}) with lots of idle time
```

         (courses, hands-ons, etc.)
\end{itemize} 

\section{Submitting jobs}
\label{sec:debug_ugent_jobs}

To submit jobs to the HPC-UGent debug cluster nicknamed \lstinline|slaking|, first use:

\begin{prompt}
%\shellcmd{module swap cluster/slaking}
\end{prompt}

Then use the familiar \lstinline|qsub|, \lstinline|qstat|, etc. commands.

\subsection{Restrictions}
\label{subsec:debug_ugent_restrictions}

> Currently one user can have maximum 5 jobs in the queue (running and waiting for run), and 
> maximum 3 jobs per user can run at the same time. The running jobs can allocate maximum 8
> CPU cores. \\ Please note that based on the (historical) workload of the debug cluster,
> the above restrictions and the overcommitment ratio might change without prior notice.

Reviewer: ```suggestion
Currently each user may have at most 5 jobs in the queue (both running and waiting to run), and 
at most 3 jobs per user can be running at the same time. The running jobs may allocate no more than 8
CPU cores in total. \\ Please note that based on the (historical) workload of the debug cluster,
the above restrictions and the overcommitment ratio might change without prior notice.
```

\chapter{HPC-UGent debug cluster}
\label{ch:debug_ugent}

\section{Purpose}
\label{sec:debug_ugent_pupose}
The purpose of the debug cluster is to run debug jobs. Additionally, it can be also used for courses. 
The debug cluster is overcommited, which means that 6 times more CPU cores can be

> scheduled than physically exist in the cluster. The cluster performance heavily depends on the
> workload, and jobs can slow down or speed up during their run. In the case of a
> fully loaded cluster, jobs would run about 6 times slower compared to an empty cluster. 

Reviewer: ```suggestion
assigned to jobs than physically exist in the cluster. Obviously, the performance of this cluster heavily depends on the workloads and that actual overcommit usage. Be aware that jobs can slow down or speed up during their execution.
```

Ditch the 6x here as well.

\\ Due to the restrictions (see section~\ref{subsec:debug_ugent_restrictions}) and sharing the CPU
resources, jobs on slaking should normally start more or less immediately. As a consequence,
typical workloads for this cluster should be limited to:
\begin{itemize}
  \item  Debugging programs
  \item  Try/debug submit scripts
  \item  Run small jobs for which performance is not an issue (courses, hands-ons, etc.)
  \item  Run interactive jobs (see chapter~\ref{ch:running-interactive-jobs}) with lots of idle time
         (courses, hands-ons, etc.) 
  \item  Run Cluster Desktops (see chapter~\ref{ch:web_portal}) with lots of idle time
         (courses, hands-ons, etc.)
\end{itemize} 

\section{Submitting jobs}
\label{sec:debug_ugent_jobs}

To submit jobs to the HPC-UGent debug cluster nicknamed \lstinline|slaking|, first use:

\begin{prompt}
%\shellcmd{module swap cluster/slaking}
\end{prompt}

Then use the familiar \lstinline|qsub|, \lstinline|qstat|, etc. commands.

> \subsection{Restrictions}

Reviewer: ```suggestion
\subsection{Restrictions and overcommit factor}
```

> \chapter{HPC-UGent debug cluster}

Reviewer 2: How about "quick start cluster", or something like that?

\label{ch:debug_ugent}

\section{Purpose}
\label{sec:debug_ugent_pupose}

> The purpose of the debug cluster is to run debug jobs. Additionally, it can be also used for courses. 

Reviewer 3: "The purpose of this cluster is to give the user an environment where there should be no waiting in the queue to get access to a limited number of resources. This environment allows a user to immediatelty start doing, and is the ideal place for interactive work such as development, debugging and light prodcution (typically sufficient for training and/or courses). This enviroment should be seen as an extension of the loginnodes, instead of the a compute resource."

some remarks:

- i would really stress the interactive part more than the debug.
- we should also recommend it for OOD/portal

> The debug cluster is overcommited, which means that more CPU cores can be

Reviewer 4: overcommi*tt*ed

typical workloads for this cluster should be limited to:
\begin{itemize}
  \item  Debugging programs
  \item  Try/debug submit scripts
  \item  Run small jobs for which performance is not an issue (courses, hands-ons, etc.)
  \item  Run interactive jobs (see chapter~\ref{ch:running-interactive-jobs}) with lots of idle time
         (courses, hands-ons, etc.) 
  \item  Run Cluster Desktops (see chapter~\ref{ch:web_portal}) with lots of idle time
         (courses, hands-ons, etc.)
\end{itemize} 

\section{Submitting jobs}
\label{sec:debug_ugent_jobs}

To submit jobs to the HPC-UGent debug cluster nicknamed \lstinline|slaking|, first use:

\begin{prompt}
%\shellcmd{module swap cluster/slaking}
\end{prompt}

Then use the familiar \lstinline|qsub|, \lstinline|qstat|, etc. commands.

\subsection{Restrictions and overcommit factor}
\label{subsec:debug_ugent_restrictions}

Currently each user may have at most 5 jobs in the queue (both running and waiting to run)
and at most 3 jobs per user can be running at the same time. The running jobs may allocate 
no more than 8 CPU cores in total. \\
At this moment, the cluster has an overcommit factor 6.  This means that 6 times more cores

> can be allocated than physically exist. Simulatenously, the default memory per core is 6

Reviewer: There is no limit atm. So they can ask all of it. We can impose a limit, afaik, but we should agree on it then.
